lib/PreconditionRunner.pm

```diff
@@ -95,12 +96,53 @@ sub write_num_passes
 sub run_to_steady_state($)
 {
     my $self = shift;
-    
     my %args = @_;
 
     my $msg_prefix = $args{'msg_prefix'};
-    my $output_dir = $args{'output_dir'};
     my $test_ref = $args{'test_ref'};
+    my $output_dir = $args{'output_dir'};
+
+    # preconditioner.exe does not support multiple files
```

Comment:
Something else is funny here.  You don't want to launch multiple precondition.exe instances, because you want them to all reach steady-state before proceeding to the test, right?  And yet, if I understand the DiskSpd alternative code, there's literally no dynamic detection of steady-state at all there -- you just run for an hour.  It seems like you've given up on steady-state detection entirely.

One way to fix this would be to add an option to precondition.exe akin to "--run_forever_even_after_reaching_steady_state", except with a much better name.  The preconditioner could print something to stdout like, "Steady state attained, send CTRL-C to terminate" or something.

StorScore would just have to launch multiple child processes of precondition.exe, wait for them to all reach steady-state, and then just kill them.